packages/typescript-estree/src/parseSettings/createParseSettings.ts

```diff
@@ -47,6 +51,13 @@ export function createParseSettings(
         : new Set(),
     errorOnTypeScriptSyntacticAndSemanticIssues: false,
     errorOnUnknownASTType: options.errorOnUnknownASTType === true,
+    EXPERIMENTAL_projectService:
+      (options.EXPERIMENTAL_useProjectService === true &&
+        process.env.TYPESCRIPT_ESLINT_EXPERIMENTAL_TSSERVER !== 'false') ||
+      (process.env.TYPESCRIPT_ESLINT_EXPERIMENTAL_TSSERVER === 'true' &&
+        options.EXPERIMENTAL_useProjectService !== false)
+        ? (TSSERVER_PROJECT_SERVICE ??= createProjectService())
+        : undefined,
```

Comment:
lol our logic for all of this is getting pretty complicated.
maybe we want to convert all these flags into an enum or something to simplify from `n` boolean flags to a more understandable representation?

Reply:
Yeah agreed, this is  ... a lot. I think it'd be good to go even further and have some kind of discriminated union for the area of project settings. That way we can be more clear on e.g. not allowing `project: './tsconfig.eslint.json'` with `useProjectService` (until/unless that becomes supported in TypeScript APIs). A good followup to investigate.